File: app/src/main/java/com/example/notes/note/papers/paper/messages/message/scraps/scrap/letters/letter/memoirs/memoir/data/notesSources/NotesSourceLocalImpl.java
package com.example.notes.note.papers.paper.messages.message.scraps.scrap.letters.letter.memoirs.memoir.data.notesSources;

import android.content.res.Resources;

import com.example.notes.note.papers.paper.messages.message.scraps.scrap.letters.letter.memoirs.memoir.R;
import com.example.notes.note.papers.paper.messages.message.scraps.scrap.letters.letter.memoirs.memoir.data.NoteData;

import java.util.ArrayList;
import java.util.List;

public class NotesSourceLocalImpl implements NotesSource {
    private List<NoteData> notes;
    private Resources resources;

    public NotesSourceLocalImpl(Resources resources) {
        notes = new ArrayList<>();
        this.resources = resources;
    }

    public NotesSourceLocalImpl(List<NoteData> notes) {
        this.notes = notes;
    }

    @Override
    public int size() {
        return notes.size();
    }

    @Override
    public NoteData getNoteData(int position) {
        return notes.get(position);
    }

    @Override
    public void addNote(NoteData note) {
        notes.add(NULL_POSITION, note);
    }

    @Override
    public void editNote(int position, NoteData note) {
        notes.set(position, note);
    }

    @Override
    public void deleteNote(int position) {
        notes.remove(position);
    }

    @Override
    public void clearAllNote() {
        notes.clear();
    }

    @Override
    public NotesSource getFavoriteData() {
        List<NoteData> favoriteData = new ArrayList<>();
        for (int i = 0; i < notes.size(); i++) {
            if (this.getNoteData(i).isFavorite()) {
                favoriteData.add(this.getNoteData(i));
            }
        }
        return new NotesSourceLocalImpl(favoriteData);
    }

    @Override
    public NotesSource init(NotesSourceResponse notesSourceResponse) {
        String[] noteTitles = resources.getStringArray(R.array.noteTitles);
        String[] notesContent = resources.getStringArray(R.array.noteContent);
        //boolean[] notesIsFavorite = resources.getBoolean(R.array.);
        for (int i = 0; i < noteTitles.length; i++) {
            NoteData.Builder noteBuilder = new NoteData.Builder()
                    .setNoteContent(notesContent[i])
                    .setTitle(noteTitles[i]);
            //.setIsFavorite();

            notes.add(noteBuilder.build());
        }
        //TODO: FavoriteList пока не может работать, нет знаний по базе данных

        if (notesSourceResponse != null) {
            notesSourceResponse.initialized(this);
        }

        return this;
    }

}

File: app/src/main/java/com/example/notes/note/papers/paper/messages/message/scraps/scrap/letters/letter/memoirs/memoir/data/notesSources/NotesSourceLocalImpl.kt
package com.example.notes.note.papers.paper.messages.message.scraps.scrap.letters.letter.memoirs.memoir.data.notesSources

import android.content.res.Resources
import com.example.notes.note.papers.paper.messages.message.scraps.scrap.letters.letter.memoirs.memoir.Const
import com.example.notes.note.papers.paper.messages.message.scraps.scrap.letters.letter.memoirs.memoir.R
import com.example.notes.note.papers.paper.messages.message.scraps.scrap.letters.letter.memoirs.memoir.data.NoteData
import java.util.*

class NotesSourceLocalImpl(
    private var resources: Resources?) : NotesSource {
    private var notes: MutableList<NoteData?> = ArrayList()

    override fun size(): Int {
        return notes.size
    }

    override fun getNoteData(position: Int): NoteData? {
        return notes[position]
    }

    override fun addNote(note: NoteData?) {
        notes.add(Const.NULL_POSITION, note)
    }

    override fun editNote(position: Int, note: NoteData?) {
        notes[position] = note
    }

    override fun deleteNote(position: Int) {
        notes.removeAt(position)
    }

    override fun clearAllNote() {
        notes.clear()
    }

    override fun init(notesSourceResponse: NotesSourceResponse?): NotesSource {
        val noteTitles = resources!!.getStringArray(R.array.noteTitles)
        val notesContent = resources!!.getStringArray(R.array.noteContent)
        for (i in noteTitles.indices) {
            val noteBuilder = NoteData.Builder()
                .setNoteContent(notesContent[i])
                .setTitle(noteTitles[i])
            notes.add(noteBuilder.build())
        }
        notesSourceResponse?.initialized(this)
        return this
    }

    private constructor(resources: Resources?, notes: MutableList<NoteData?>) : this(resources = resources){
        this.notes = notes
    }

    override fun copy(): NotesSource {
        return NotesSourceLocalImpl(resources, notes)
    }
}